integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/e2e/openstack/OpenStackTests.java

```diff
@@ -0,0 +1,228 @@
+            String[] groups,
+            String scaleGroup,
+            @Description TestCaseDescription description) {
+
+        testContext
+                .given(ClusterTestDto.class)
+                .withClusterDefinitionName(clusterDefinitionName)
+                .given(StackTestDto.class)
+                .withInstanceGroupsEntity(getInstanceGroupTestDtos(testContext, groups))
+                .withCluster(ClusterTestDto.class.getSimpleName())
+                .when(stackTestClient.createV4())
+                .await(STACK_AVAILABLE)
+                .when(stackTestClient.scalePostV4()
+                        .withGroup(scaleGroup)
+                        .withDesiredCount(2))
+                .await(STACK_AVAILABLE)
+                .when(stackTestClient.scalePostV4()
+                        .withGroup(scaleGroup)
+                        .withDesiredCount(1))
+                .await(STACK_AVAILABLE)
+                .when(stackTestClient.stopV4())
+                .await(STACK_STOPPED)
+                .when(stackTestClient.startV4())
+                .await(STACK_AVAILABLE)
+                .then((tc, testDto, cc) -> stackTestClient.deleteV4().action(tc, testDto, cc))
+                .validate();
+    }
+
+    @DataProvider(name = "openStackHdfTestsDataProvider")
+    public Object[][] openStackHdfTestsDataProvider() {
+        SparklessTestContext testContext = getBean(SparklessTestContext.class);
+        minimalSetupForClusterCreation(testContext);
+
+        Object[][] data = new Object[getHdfClusterDefinition().length][5];
+        for (int i = 0; i < getHdfClusterDefinition().length; i++) {
+            data[i][0] = testContext;
+            data[i][1] = getHdfClusterDefinition()[i];
+            data[i][2] = getHdfGroups();
+            data[i][3] = getHdfScaleGroup();
+            data[i][4] = new TestCaseDescription.TestCaseDescriptionBuilder()
+                    .given("there is an available cloudbreak")
+                    .when("a stack create request with "
+                            + getHdfClusterDefinition()[i] + " cluster definition is sent AND scale AND stop AND start requests are sent")
+                    .then("all stack operation should succeed and the stack should be deletable");
+        }
+        if (isHdfEnabled()) {
+            return data;
+        }
+        return new Object[0][5];
+    }
+
+    private List<InstanceGroupTestDto> getInstanceGroupTestDtos(TestContext testContext, String[] groups) {
+        List<InstanceGroupTestDto> instanceGroups = new ArrayList<>();
+        for (int i = 0; i < groups.length; i++) {
+            HostGroupType hostGroupType = HostGroupType.getByName(groups[i]);
+            instanceGroups.addAll(InstanceGroupTestDto.withHostGroup(testContext, hostGroupType));
+        }
+        for (InstanceGroupTestDto instanceGroup : instanceGroups) {
+            if (HostGroupType.WORKER.getName().equals(instanceGroup.getName())) {
```

Comment:
Move this `if` to the previous for loop, and then this second loop becomes unnecessary. By the way with this solution every cluster will start with 3 worker nodes.

Reply:
I dont see any problem with 3 worker